invoke-ai/InvokeAI
# Modify title and favicon when item in progress (issue #2392)

## Summary

### Description

Two methods (`updatePageTitle` and `updatePageFavicon`) were created and executed within the WebSocket handler.

When a Queue item's status changes, the methods append a `(1) ` prefix and modify the favicon, respectively. I included an `id` attribute to the `link` element to ensure the query wouldn't fail or collide.

After Hipsterusername's suggestion, I modified the `updatePageTitle` method to display the active queue length instead of (1). 

### Files changed:
1. `invokeai/frontend/web/src/app/store/middleware/listenerMiddleware/listeners/socketio/socketQueueItemStatusChanged.ts`
2. `invokeai/frontend/web/index.html`

### Files added
1. `invokeai/frontend/web/public/assets/images/invoke-alert-favicon.svg`

### Demo

https://github.com/invoke-ai/InvokeAI/assets/12788435/d60db7f9-0e7e-4204-9914-ab968e03f165

(I will update the final demo video with all suggestions implemented once it's done)

### Open questions

1. Do you have any recommendations on how to create automated tests?
3. Should I split the methods into a dedicated file and import them on the Socket handler? I thought about doing so, but it would make more sense to go with the simplest approach and adjust if needed.

## Related Issues / Discussions

Address issue #2392 

## QA Instructions

1. Ensure there is an active model
4. Input a prompt
5. Click on the `invoke` button
6. Observe the title and favicon

## Merge Plan

-

## Checklist
